### PR TITLE
Resize fixes

### PR DESCRIPTION
**Pull request by dwlehman**
These should resolve issues with resize -- particularly when the volume has an existing file system.

Still needed:
- test validation of volume sizes
- existing resize test(s) re-enabled

### CODE REVIEW

File: library/blivet.py [hunk 1]
```diff
@@ -231,7 +231,17 @@ def _resize(self):
         except Exception:
             raise BlivetAnsibleError("invalid size specification for volume '%s': '%s'" % (self._volume['name'], self._volume['size']))
 
-        if size and self._device.resizable and self._device.size != size:
+        if size and self._device.size != size:
+            try:
+                self._device.format.update_size_info()
+            except FSError:
+                raise BlivetAnsibleError("volume '%s' cannot be resize because it needs repair" % self._device.name)
```
Comment (vojtechtrefny) [inline library/blivet.py:238]: I think this isn't necessary, `update_size_info` shouldn't raise FSError anywhere now.

**Comment by pcahyna** [comment]
Are there tests for the bugs that this fixes? Or is it enough to enable the currently disabled tests?

**Comment by pcahyna** [comment]
will it fix this problem in rebased #64 : https://github.com/linux-system-roles/storage/pull/106#issuecomment-644316059 "volume 'luks-vdb1' cannot be resized from 4 GiB to 4 GiB" ?

**Comment by yizhanglinux** [comment]
@dwlehman @pcahyna 
This pull fixed ext2/3/4 resize function on RHEL7/8, pls help review and merge it.

File: library/blivet.py (hunk 1 above)
Comment (pcahyna) [reply library/blivet.py:238]: @dwlehman @japokorn  should this comment be addressed?

**Comment by pcahyna** [comment]
@dwlehman I have a version updated to master, I can rebase the branch if you want.

**Comment by pcahyna** [comment]
sorry, I see you already did so

**Comment by dwlehman** [comment]
@pcahyna ok to merge?

**Comment by pcahyna** [comment]
@dwlehman can you please add a test from https://github.com/pcahyna/storage/tree/resize-fixes ? I was not able to push to your branch.
With the test it is ok to merge.

**Pull request merged**